server/service/mdm.go

```diff
@@ -1113,3 +1115,233 @@ func isAppleProfileID(profileIDOrUUID string) (uint, bool) {
 	}
 	return uint(id), true
 }
+
+////////////////////////////////////////////////////////////////////////////////
+// Batch Replace MDM Profiles
+////////////////////////////////////////////////////////////////////////////////
+
+type batchSetMDMProfilesRequest struct {
+	TeamID   *uint             `json:"-" query:"team_id,optional"`
+	TeamName *string           `json:"-" query:"team_name,optional"`
+	DryRun   bool              `json:"-" query:"dry_run,optional"` // if true, apply validation but do not save changes
+	Profiles map[string][]byte `json:"profiles"`
+}
+
+type batchSetMDMProfilesResponse struct {
+	Err error `json:"error,omitempty"`
+}
+
+func (r batchSetMDMProfilesResponse) error() error { return r.Err }
+
+func (r batchSetMDMProfilesResponse) Status() int { return http.StatusNoContent }
+
+func batchSetMDMProfilesEndpoint(ctx context.Context, request interface{}, svc fleet.Service) (errorer, error) {
+	req := request.(*batchSetMDMProfilesRequest)
+	if err := svc.BatchSetMDMProfiles(ctx, req.TeamID, req.TeamName, req.Profiles, req.DryRun, false); err != nil {
+		return batchSetMDMProfilesResponse{Err: err}, nil
+	}
+	return batchSetMDMProfilesResponse{}, nil
+}
+
+func (svc *Service) BatchSetMDMProfiles(ctx context.Context, tmID *uint, tmName *string, profiles map[string][]byte, dryRun, skipBulkPending bool) error {
+	var err error
+	if tmID, tmName, err = svc.authorizeBatchProfiles(ctx, tmID, tmName); err != nil {
+		return err
+	}
+
+	appCfg, err := svc.ds.AppConfig(ctx)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "getting app config")
+	}
+
+	if err := validateProfiles(profiles); err != nil {
+		return ctxerr.Wrap(ctx, err, "validating profiles")
+	}
+
+	appleProfiles, err := getAppleProfiles(ctx, tmID, appCfg, profiles)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "TODO")
+	}
+
+	windowsProfiles, err := getWindowsProfiles(ctx, tmID, appCfg, profiles)
+	if err != nil {
+		return ctxerr.Wrap(ctx, err, "TODO")
```

Comment:
TODO to replace with actual context information.

Reply:
thank you very much and sorry about that! completely missed it!